### PR TITLE
[next] WIP: Fixes #417 but does not cater for #72.

### LINKED ISSUE #417: Display parameters for storage queries
Bring it back, currently missing after API upgrades for PoC-3 - https://github.com/polkadot-js/apps/pull/416/files#diff-b548c114245b6470a658f3c2840c5a26R58

### PR DESCRIPTION
**Pull request by ltfschoen**
* Fixes #417 (well it appears to fix it as shown in screenshot below), but only caters for Chain State queries with 0 or 1 parameter,
for >=2 parameters to be catered for we need to address #72

screen shot 2018-11-08 at 11 37 47

* Incomplete is resolving TSLint errors
(i.e. `Argument of type 'RawParam & TypeDef' is not assignable to parameter of type 'MethodParams'.
Property 'params' is missing in type 'RawParam & TypeDef'.`) when trying to combine the method
parameter's value and type into a single object (since the method parameter's value and type are
passed into the `onAdd` separately contained in the `values` and `params` of the `this.state`)

* Incomplete is knowing whether to apply the `isValid` property further.

* Incomplete is checking the functionality is correct in the "Raw key" tab of http://localhost:3000/#/chainstate
(where a hex-encoded storage key is provided) since functionality has only been tested in the "Module" tab.

### CODE REVIEW

**Comment by jacogr** [comment]
Atm all the params are either only 0 or 1 (in this case of params for a map), so we will be safe for the time being

**Comment by ltfschoen** [comment]
there's actually one that takes two parameters `councilVoting > councilVoteOf((Hash, AccountId)): bool`

**Comment by jacogr** [comment]
Ahhh, well, that is actually still only 1 param, it is a Tuple. (Long-standing issue logged around those for support as well, same as Arrays, Params don't allow input yet)

File: packages/app-storage/src/Query.tsx [hunk 1]
```diff
@@ -53,16 +53,16 @@ class Query extends React.PureComponent<Props, State> {
...
-      //   </span>
-      // ));
-      : [];
+    const { params } = value;
+    // FIXME - we need to cater for multiple params. see code prior to this commit
+    // FIXME - we haven't used params.isValid, should we?
+    // i.e. councilVoting > councilVoteOf(Hash, AccountId)
+    const inputs = params.map(({ isValid, type, value }) => (
+        <span key={`param_${type}`}>
+          {type}={valueToText(type, params[0].value)}
```
Comment (jacogr) [inline packages/app-storage/src/Query.tsx]: So not sure here why we refer to `params[0]` when mapping through `params` - value is exploded

File: packages/ui-app/src/Params/types.d.ts [hunk 2]
```diff
@@ -16,6 +17,7 @@ export type RawParam$Values = RawParam$Value | RawParam$ValueArray;
 
 export type RawParam = {
   isValid: boolean,
+  key: StorageFunction,
```
Comment (jacogr) [inline packages/ui-app/src/Params/types.d.ts]: This is not always the case, e.g. for Extrinsics, RawParam is used and it is not storage at all.

File: packages/app-storage/src/Selection/Modules.tsx [hunk 3]
```diff
@@ -101,11 +110,18 @@ class Modules extends React.PureComponent<Props, State> {
...
+    // if the method has a parameter, store the value and
+    //  type of the method's parameter in the same object
+    if (values.length && params.length) {
+      newParams.push(Object.assign({}, values[0], params[0]['type']));
+    }
 
     onAdd({
       key,
-      params: values
+      params: newParams
```
Comment (jacogr) [inline packages/app-storage/src/Selection/Modules.tsx]: It would seem like all this bubbles up now, as opposed to just the `isValid` & `value`?

**Comment by jacogr** [comment]
Closing as stale. Re-open as required.

**Comment by ltfschoen** [comment]
replaced by #496

**Pull request closed without merging**